src/System.Security.Cryptography.Xml/tests/Samples/EncryptingDecryptingAsymmetric.cs

```diff
@@ -0,0 +1,103 @@
+<test>some text node</test>
+</example>";
+
+        private static XmlDocument LoadXmlFromString(string xml)
+        {
+            var doc = new XmlDocument();
+            doc.PreserveWhitespace = true;
+            doc.LoadXml(xml);
+            return doc;
+        }
+
+        private static void Encrypt(XmlDocument doc, string elementName, string encryptionElementID, RSA rsaKey, string keyName)
+        {
+            var elementToEncrypt = (XmlElement)doc.GetElementsByTagName(elementName)[0];
+
+            using (var sessionKey = Aes.Create())
+            {
+                sessionKey.KeySize = 256;
+
+                // Encrypt the session key and add it to an EncryptedKey element.
+                var encryptedKey = new EncryptedKey()
+                {
+                    CipherData = new CipherData(EncryptedXml.EncryptKey(sessionKey.Key, rsaKey, false)),
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncRSA15Url)
+                };
+
+                // Specify which EncryptedData
+                // uses this key. An XML document can have
+                // multiple EncryptedData elements that use
+                // different keys.
+                encryptedKey.AddReference(new DataReference()
+                {
+                    Uri = "#" + encryptionElementID
+                });
+
+                var encryptedData = new EncryptedData()
+                {
+                    Type = EncryptedXml.XmlEncElementUrl,
+                    Id = encryptionElementID,
+
+                    // Create an EncryptionMethod element so that the
+                    // receiver knows which algorithm to use for decryption.
+                    EncryptionMethod = new EncryptionMethod(EncryptedXml.XmlEncAES256Url)
+                };
+
+                encryptedData.KeyInfo.AddClause(new KeyInfoEncryptedKey(encryptedKey));
+                encryptedKey.KeyInfo.AddClause(new KeyInfoName()
+                {
+                    Value = keyName
+                });
+
+                var encryptedXml = new EncryptedXml();
+                encryptedData.CipherData.CipherValue = encryptedXml.EncryptData(elementToEncrypt, sessionKey, false);
+
+                EncryptedXml.ReplaceElement(elementToEncrypt, encryptedData, false);
+            }
+        }
+
+        public static void Decrypt(XmlDocument doc, RSA rsaKey, string keyName)
+        {
+            var encrypted = new EncryptedXml(doc);
+            encrypted.AddKeyNameMapping(keyName, rsaKey);
+            encrypted.DecryptDocument();
+        }
+
+        [Fact]
+        public void AsymmetricEncryptionRoundtrip()
+        {
+            using (RSA key = RSA.Create())
+            {
+                XmlDocument xmlDocToEncrypt = LoadXmlFromString(ExampleXml);
+                Encrypt(xmlDocToEncrypt, ExampleXmlRootElement, "EncryptedElement1", key, "rsaKey");
+
+                Assert.DoesNotContain("some text node", xmlDocToEncrypt.OuterXml);
```

Comment:
@krwq I'm just always paranoid when it comes to tests that would pass if no work was done.

Pseudo-cody:
```C#
const string TestString = "some text node";

XmlDocument xmlDocToEncrypt = ...;
// Make sure that the string we're making sure got encrypted is already there.
Assert.Contains(TestString, xmlDocToEncrypt.OuterXml);
Encrypt();
// Remember to check that this existed before calling Encrypt(),
// otherwise this could accidentally pass due to not existing in the input document at all.
Assert.DoesNotContain(TestString, xmlDocToEncrypt.OuterXml);
Decrypt();
Assert.Equal(originalDocument, decryptedDocument);
```

What you want to test is "this was here, now it's gone"; so the positive test ensures that "this is here", and then the negative test asserts "now it's gone".